### ISSUE TITLE
imgkit with xvfb renders blank images in output file

### ISSUE THREAD

**Issue by estherbouquet** [jarrekk/imgkit#55]
hi!
I am using a raspberry 3 B+ running Stretch Version 9.

I developed a python program using imgkit. I have html files in a local directory and I am trying to output them in jpg files.
When I launch my program manually from the terminal, my program works perfectly fine: my html files (that contains text and jpg files) are rendered with the correct css and saved as a jpg file with this message:
```
Loading page (1/2)
Rendering (2/2)
Done
```
But when I tried the first time to launch my script with `systemctl start name.service` , I got an error message telling me 
```
QXcbConnection: Could not connect to display
You need to install xvfb(sudo apt-get install xvfb, then add option: {"xvfb": ""}
```
I followed the instructions, restarted my service, the program launched without the previous error. This is where the problem happens: when I looked at the output files, the css is correct, the text also, but all my images are blank. 
I also have this error poping into my terminal:

```
libEGL warning: DRI2: failed to create any config
libEGL warning: DRI2: failed to create any config
Loading page (1/2)
Rendering (2/2)
Done
```
After looking all around the web, I can't find anything close to an answer. Do you have any idea of what to do?
Thank you 🌻

**Comment by jarrekk** [comment]
@estherbouquet thanks for the question.
```
libEGL warning: DRI2: failed to create any config
libEGL warning: DRI2: failed to create any config
```
Looks this error is about the raspberry pi, can you try on another environment like an Ubuntu server?

**Comment by estherbouquet** [comment]
Unfortunately I would not know how to do it, my knowledge is limited, sorry.

**Comment by jarrekk** [comment]
How about running `wkhtmltoimage` directly and check the result? Did you install `wkhtmltopdf` on raspberry pi?

**Comment by estherbouquet** [comment]
I'm going to try and I'll let you know. thanks!

**Comment by jarrekk** [comment]
> Yes I install wkhtmltopdf by using apt-get install wkhtmltopdf

I am not sure if you install the package for raspbian pi platform, there are multiple packages in one release. https://github.com/wkhtmltopdf/wkhtmltopdf/releases/tag/0.12.5

**Comment by estherbouquet** [comment]
> How about running `wkhtmltoimage` directly and check the result? Did you install `wkhtmltopdf` for raspberry pi platform?

So I tried to run `wkhtmltoimage` in my terminal and no problem, the pictures are here even though I have `libEGL warning: DRI2: failed to authenticate` showing up. When I do `xvfb-run wkhtlmtoimage` in my terminal, no picture are rendered though.

After trying that, I reinstalled the `wkhtmltopdf` package for the stretch version (wkhtmltox_0.12.5-1.stretch_amd64.deb) instead of bionic as in your readme and no luck there neither. Still no picture and still the same error with libEGL and DRI2 anytime I use xvfb with imgkit or xvfb-run with wkhtmltoimage.

Do you think it's because I need to give a display number to xvfb somewhere in order for the raspberry to know which display screen to use to render the pictures?

**Comment by estherbouquet** [comment]
hi again! I think I've found the solution after all and you led me to this, so thank you so much!

You were right, it was coming from the `wkhtmltopdf` that was installed when I first did (months ago) `sudo apt-get install wkhtmltopdf`. Definitely a wrong move. I realised that when I did a `wkhtmltopdf -V` after installing `wkhtmltox_0.12.5-1.stretch_amd64.deb`: it was not using this new version.
I also found out about amd64 and i386: I didn't know Stretch could be rather 64bit or 32. In my case, using `uname -m` it gave me back `armv7l` which is... 32bit. The amd64 version that I downloaded was completely useless. 

So I did `sudo apt-get purge --auto-remove wkhtmltopdf` to get rid of the package and its dependencies and then went on the [wkhtmltopdf release page](https://wkhtmltopdf.org/downloads.html) to copy the link for the Debian 9 (Stretch) raspberrypi architecture.

I followed the instructions of your [init.sh file](https://github.com/jarrekk/imgkit/blob/master/travis/init.sh), replacing `https://github.com/wkhtmltopdf/packaging/releases/download/0.12.6-1/wkhtmltox_0.12.6-1.bionic_amd64.deb` with the link I've just copied for Stretch.

When I did `wkhtmltopdf -V` to see if I had the right version this time, it told me that it couldn't find any file or folder. With `which wkhtmltopdf` I was able to see that I installed it in `/usr/local/bin/` instead of `/usr/bin/` so I moved the file back using `mv /usr/local/bin/wkhtmltopdf /usr/bin`. 
`wkhtmltopdf -V` worked like a charm this time and returned `wkhtmltopdf 0.12.6 (with patched qt)`.

Full of hope, I tried `wkhtmltoimage inputfile outputfile` and had a new error: `Warning: Blocked access to file` which was easily solved by doing `wkhtmltoimage --enable-local-file-access inputfile outputfile` as explained [here](https://stackoverflow.com/questions/62315246/wkhtmltopdf-0-12-6-warning-blocked-access-to-file). Then, when I tried adding `xvfb-run` just before this line, I had no error and my images were finally pictured! 🥳 

I tried to do the same in my python code by doing: 
``` 
options={
  'enable-local-file-access': '', 
  'xvfb':""
}

imgkit.from_file(inputfile, outputfile, options=options) 
```

And it also worked 😭 I can't believe it! 
Thanks again for your support and I promised, I'll stop cursing xvfb next time haha
